arangod/Cluster/ClusterIndexMethods.cpp

```diff
@@ -932,61 +854,66 @@ auto ensureIndexCoordinatorReplication2Inner(
+    auto waitOnSuccess =
+        callbackRegistry
+            .waitFor(
+                pathCollectionInPlan(collection.vocbase().name(),
+                                     std::to_string(collection.id().id()))
+                    ->str(arangodb::cluster::paths::SkipComponents(1)),
+                [id = std::string{idString}](VPackSlice slice) {
+                  if (slice.isNone()) {
+                    // TODO: Should this actual set an "error"? It indicates
+                    // that the collection is dropped if i am not mistaken
+                    return false;
+                  }
+                  auto collection = velocypack::deserialize<
+                      replication2::agency::CollectionPlanSpecification>(slice);
+                  auto const& indexes = collection.indexes.indexes;
+                  for (auto const& index : indexes) {
+                    auto indexSlice = index.slice();
+                    if (indexSlice.hasKey(StaticStrings::IndexId) &&
+                        indexSlice.get(StaticStrings::IndexId)
+                            .isEqualString(id)) {
+                      if (!indexSlice.hasKey(StaticStrings::IndexIsBuilding)) {
+                        return true;
+                      }
+                    }
+                  }
+                  return false;
+                })
+            .thenValue([&clusterInfo](auto index) {
+              return clusterInfo.waitForPlan(index);
+            });
+    auto res = waitOnSuccess.get();
+    if (res.fail()) {
+      return res;
     }
-    // We do not have a final result. Let's wait for more input
-    // Wait for the next incomplete callback
-    for (auto& [cb, indexId] : callbackList) {
-      if (!report->hasReported(indexId)) {
-        // We do not have result for this collection, wait for it.
-        bool gotTimeout;
-        {
-          // This one has not responded, wait for it.
-          std::unique_lock guard(cb->_cv.mutex);
-          gotTimeout = cb->executeByCallbackOrTimeout(pollInterval);
-        }
-        if (gotTimeout) {
-          // We got woken up by waittime, not by  callback.
-          // Let us check if we skipped other callbacks as well
-          for (auto& [cb2, cid2] : callbackList) {
-            if (!report->hasReported(cid2)) {
-              // Only re check those where we have not yet found a
-              // result.
-              cb2->refetchAndUpdate(true, false);
-            }
-          }
-        }
-        // Break the callback loop,
-        // continue on the check if we completed loop
-        break;
-      }
+    VPackBuilder resultBuilder;
+    {
+      VPackObjectBuilder guard(&resultBuilder);
+      // add all original values
+      resultBuilder.add(VPackObjectIterator(newIndexBuilder.slice()));
+      resultBuilder.add("isNewlyCreated", VPackValue(true));
     }
+    // Need to wait here, until ClusterInfo has updated to latest plan.
```

Comment:
Are we waiting implicitly, as a consequence of the above callback?
Or, is this something that should be added in a separate PR?

Reply:
The comment is in the wrong line .... Will fix